src/argosim/metrics_utils.py

```diff
@@ -98,3 +102,277 @@ def compute_metrics(img1, img2):
+    x0, y0 : float
+        Center of the Gaussian.
+
+    Returns
+    -------
+    np.ndarray
+        Flattened 2D elliptical Gaussian evaluated at coords.
+    """
+    x, y = coords
+    x0, y0 = float(x0), float(y0)
+
+    xp = (x - x0) * np.cos(theta) + (y - y0) * np.sin(theta)
+    yp = -(x - x0) * np.sin(theta) + (y - y0) * np.cos(theta)
+
+    return (
+        amp * np.exp(-((xp**2) / (2 * sigma_x**2) + (yp**2) / (2 * sigma_y**2)))
+        + offset
+    )
+
+
+def get_beam_crop(beam, center, crop_size):
+    """
+    Extract a square region from the beam centered at a given pixel.
+
+    Parameters
+    ----------
+    beam : np.ndarray
+        Full 2D beam image.
+    center : tuple
+        Pixel coordinates (x, y) of the beam center.
+    crop_size : int
+        Size of the square region to extract.
+
+    Returns
+    -------
+    np.ndarray
+        Cropped square region of shape (crop_size, crop_size).
+    """
+    x0, y0 = center
+    half = crop_size // 2
+    x1, x2 = x0 - half, x0 + half
+    y1, y2 = y0 - half, y0 + half
+    return beam[y1:y2, x1:x2]
+
+
+def fit_elliptical_gaussian_to_crop(crop, x0_local, y0_local):
+    """
+    Fit an elliptical 2D Gaussian model to a cropped beam region.
+
+    Parameters
+    ----------
+    crop : np.ndarray
+        2D cropped beam region.
+    x0_local, y0_local : int
+        Local coordinates of the Gaussian center in the crop.
+
+    Returns
+    -------
+    popt : list
+        Optimal parameters [amp, sigma_x, sigma_y, theta, offset].
+    """
+    y, x = np.indices(crop.shape)
+
+    def model(coords, amp, sigma_x, sigma_y, theta, offset):
+        return gaussian_2d_elliptical(
+            coords, amp, sigma_x, sigma_y, theta, offset, x0_local, y0_local
+        )
+
+    p0 = (np.max(crop), 4.5, 4.5, 0, 0)
+    bounds = ([0, 0.5, 0.5, -np.pi, -np.inf], [np.inf, 10, 10, np.pi, np.inf])
+
+    popt, _ = curve_fit(
+        model, (x.ravel(), y.ravel()), crop.ravel(), p0=p0, bounds=bounds
+    )
+    return popt
+
+
+def compute_fwhm(sigma_x, sigma_y):
+    """
+    Convert standard deviations to FWHM.
+
+    Parameters
+    ----------
+    sigma_x : float
+        Standard deviation along x-axis.
+    sigma_y : float
+        Standard deviation along y-axis.
+    Returns
+    -------
+    tuple
+        FWHM along x and y.
+    """
+    factor = 2.553  # approximation for FWHM = 2*sqrt(2*ln(2))
+    return factor * sigma_x, factor * sigma_y
+
+
+def compute_eccentricity(fwhm_x, fwhm_y):
+    """
+    Compute the eccentricity of an ellipse from FWHM values.
+
+    Parameters
+    ----------
+    fwhm_x : float
+        FWHM along x-axis.
+    fwhm_y : float
+        FWHM along y-axis.
+
+    Returns
+    -------
+    float
+        Ellipticity e ∈ [0, 1], where 0 = circle and 1 = highly elongated.
+    """
+    a = max(fwhm_x, fwhm_y)
+    b = min(fwhm_x, fwhm_y)
+    return np.sqrt(1 - (b / a) ** 2)
+
+
+def fit_elliptical_beam(beam, center=(271, 271), crop_size=4):
```

Comment:
The name of the function is a bit misleading. Even if it does fit an elliptical beam, it is the function to get the FWHM metrics given a radio beam.

Reply:
If the aim of this function is just to fit the gaussian, then a the function for computing the fwhm metric is missing.

Reply:
You have to think in the interaction of the user with these functions. What we want is a function `compute_fwhm(beam)` which takes the beam image and outputs the fwhm in pixels (or degrees if we give the FOV size as well). 

The current `compute_fwhm` takes as arguments the values `sigma_x` and `sigma_y` which the user won't have (same for the `compute_eccentricity` function). 

You can add a sort of wrapper function which calls every intermediate step to get the fwhm, eccentricity, ssl, etc. In such a way that the user calls `compute_beam_metrics(beam)` and gets all the relevant metrics, without having to run every intermediate step.